src/mock.js

```diff
@@ -82,6 +82,17 @@ function addMockFunctionsToSchema({ schema, mocks = {}, preserveResolvers = fals
     return customMock;
   }
 
+  function assignResolveType(type) {
+    const fieldType = getNullableType(type);
+    const namedFieldType = getNamedType(fieldType);
+
+    if (namedFieldType instanceof GraphQLUnionType ||
+        namedFieldType instanceof GraphQLInterfaceType
+    ) {
+      namedFieldType.resolveType = (data, context, info) => info.schema.getType(data.typename);
```

Comment:
what do you think is the best way to check that? I can't do `if (!namedFieldType.resolveType)` because resolveType is not undefined (the default one just returns null).